selenium_test.py

```diff
@@ -0,0 +1,235 @@
+    endDay.select_by_value('01')  # select by value
+
+    # select end month from drop down
+    endMonth = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_toMonth"]'))
+    endMonth.select_by_value('01')  # select by value
+
+    # select end Year from drop down
+    endYear = Select(driver.find_element_by_xpath('//*[@id="dateLimiterValue_da_toYear"]'))
+    endYear.select_by_value(endyear)  # select by value
+
+    # clicks Search Submit button
+    searchSubmit = driver.find_element_by_xpath("/html/body/div[3]/form/div[1]/div[2]/a")
+    searchSubmit.click()
+
+    # calls collect results on search results
+    collectResults(driver, keyword)
+    enabledButton = driver.find_elements(By.XPATH, '//*[@title="Next"]')
+    hasNextPage = False
+    if len(enabledButton) > 0:
+        hasNextPage = True
+
+    while hasNextPage:
+        disabledButton = driver.find_elements(By.XPATH, "/html/body/div[4]/div[3]/form/div[3]/div/ul[2]/b")
+        enabledButton = driver.find_elements(By.XPATH, '//*[@title="Next"]')
+        print("Length of enabled button" + str(len(enabledButton)))
+        print("Length of diabled button" + str(len(disabledButton)))
+
+        if len(enabledButton) > 0:
+            enabledButton[0].click()
+            collectResults(driver, keyword)
+            hasNextPage = True
+        elif len(disabledButton) > 0:
+            collectResults(driver, keyword)
+            break
+
+    # waits for the new rollover download
+    markedWait = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.CLASS_NAME, "markListSpan"))
+    )
+
+    # clicks marked items button
+    markedItems = driver.find_element_by_class_name("markListSpan")
+    markedItems.click()
+
+    # clicks marked items button
+    download = driver.find_element_by_id("download")
+    download.click()
+
+    currentWindow = driver.current_window_handle
+    downloadPage = driver.window_handles[1]
+    driver.switch_to.window(downloadPage)
+
+    # waits for the new rollover download
+    resultsWait = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.CLASS_NAME, "rolloverdownload"))
+    )
+
+    # clicks marked items button
+    rolloverdownload = driver.find_element_by_class_name("rolloverdownload")
+    rolloverdownload.click()
+
+    driver.switch_to.window(currentWindow)
+
+    title_list = []
+    description_list = []
+    category_list = []
+    keyword_list = []
+    link_list = []
+
+    for obj in allResults:
+        title_list.append(obj.title)
+        description_list.append(obj.description)
+        category_list.append(obj.category)
+        keyword_list.append(obj.keyword)
+        link_list.append(obj.link)
+
+
+    df = pd.DataFrame(list(zip(title_list, description_list, category_list, keyword_list, link_list)),
+                      columns=['Title', 'Description', 'Category', 'Keyword', "link"])
+
+    df.to_csv('selenium-output.csv', index=True)
```

Comment:
I don't think you need multiple lists even here - take a look at pandas.DataFrame.from_records: https://stackoverflow.com/a/41762270